### PR TITLE
Add label template functions

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Add the documentation for the template functions for labels, added in https://github.com/home-assistant/core/pull/111024

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/111024
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -574,6 +574,59 @@ If there is more than one entry with the same title, the entities for all the ma
 
 {% endraw %}
 
+### Labels
+
+- `labels()` returns the full list of labels IDs, or those for a given area ID, device ID, or entity ID.
```
Comment (c0ffeeca7) [inline source/_docs/configuration/templating.markdown]:
```suggestion
- `labels()` returns the full list of label IDs, or those for a given area ID, device ID, or entity ID.
```

File: source/_docs/configuration/templating.markdown [hunk 2]
```diff
@@ -574,6 +574,59 @@ If there is more than one entry with the same title, the entities for all the ma
...
+- `label_id(lookup_value)` returns the label ID for a given label name.
+- `label_name(lookup_value)` returns the label name for a given label ID.
+- `label_areas(label_name_or_id)` returns the list of areas IDs tied to a given label ID or name.
```
Comment (c0ffeeca7) [inline source/_docs/configuration/templating.markdown]:
```suggestion
- `label_areas(label_name_or_id)` returns the list of area IDs tied to a given label ID or name.
```

File: source/_docs/configuration/templating.markdown [hunk 3]
```diff
@@ -574,6 +574,59 @@ If there is more than one entry with the same title, the entities for all the ma
...
+- `label_devices(label_name_or_id)` returns the list of device IDs tied to a given label ID or name.
+- `label_entities(label_name_or_id)` returns the list of entity IDs tied to a given label ID or name.
+
+Each of the label template functions, can also be used as a filter.
```
Comment (c0ffeeca7) [inline source/_docs/configuration/templating.markdown]:
```suggestion
Each of the label template functions can also be used as a filter.
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @frenck :+1:

File: source/_docs/configuration/templating.markdown [hunk 4]
```diff
@@ -574,6 +574,59 @@ If there is more than one entry with the same title, the entities for all the ma
...
+```text
+{{ label_name('energy_saver') }}  # 'Energy saver'
+```
+
+```text
+{{ label_areas('security') }}  # ['driveway', 'garden', 'porch']
+```
+
+```text
+{{ label_devices('energy_saver') }}  # ['deadbeefdeadbeefdeadbeefdeadbeef']
```
Comment (c0ffeeca7) [inline source/_docs/configuration/templating.markdown:621]: deadbeef: is this intended? or was it a place holder?

Comment (frenck) [reply source/_docs/configuration/templating.markdown:621]: It is a common placeholder on the page for device IDs 🤷

**Pull request merged**